viewing_party/party.py

```diff
@@ -1,23 +1,141 @@
 # ------------- WAVE 1 --------------------
 
+from itertools import count
+
+
 def create_movie(title, genre, rating):
-    pass
+    if None in [title, genre, rating]:
+        return None
+
+    movie_dict = {
+                "title" : title,
+                "genre" : genre,
+                "rating" : rating
+    }
+    return movie_dict
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for i in range(len(user_data["watchlist"])):
+        if user_data["watchlist"][i]["title"] == title:
+            just_watched = user_data["watchlist"].pop(i)
+            user_data["watched"].append(just_watched) 
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0.00
+
+    if len(user_data["watched"]) == 0:
+        return sum
+
+    for movie in user_data["watched"]:
+        sum += movie["rating"]
+    average = sum / len(user_data["watched"])
+    return average
+
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+    max_genre = ["", 0]
+
+    if len(user_data["watched"]) == 0:
+        return None
+
+    for current in user_data["watched"]:
+        if current["genre"] in genre_dict:
+            genre_dict[current["genre"]] += 1
+        else:
+            genre_dict[current["genre"]] = 1
+        if max_genre[1] < genre_dict[current["genre"]]: #checks for a new high score(most popular genre)
+            max_genre[0] = current["genre"]
+            max_genre[1] = genre_dict[current["genre"]]
+    return max_genre[0]
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    movie_list = user_data["watched"]
+    friend_list = combine_friend_movies(user_data)
+    user_only =[]

```

Comment:
```python
    user_only = []
```

There should be space between operators and values as according to Python style guide pep8 https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements